src/stores/AppStore.ts

```diff
@@ -1303,12 +1303,40 @@ export class AppStore {
                 await this.loadDefaultFiles();
                 this.setCursorFrozen(this.preferenceStore.isCursorFrozen);
                 this.updateASTColors();
+                if (this.preferenceStore.checkNewRelease) {
+                    this.checkNewRelease();
+                }
             } catch (err) {
                 console.error(err);
             }
         }
     };
 
+    private checkNewRelease = () => {
+        fetch("https://api.github.com/repos/CARTAvis/carta/releases", {headers: {Accept: "application/vnd.github+json"}})
+            .then(response => {
+                if (!response.ok) {
+                    throw new Error(response.statusText);
+                }
+                return response.json();
+            })
+            .then(async releases => {
+                const currentRelease = "v" + CARTA_INFO.version;
+                const latestRelease = releases[0].tag_name;
```

Comment:
need to use `?.[0]?.tag_name` in case the releases JSON file is not the expected format

Reply:
fixed